src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -524,6 +527,15 @@ public boolean checkPathExists(@NonNull String branchOrHash, @NonNull String pat
             // https://support.atlassian.com/bitbucket-cloud/docs/use-bitbucket-rest-api-version-1/
         } else if (HttpStatus.SC_NOT_FOUND == status || HttpStatus.SC_UNAUTHORIZED == status) {
             return false;
+        } else if (API_RATE_LIMIT_CODE == status) {
+            LOGGER.fine("Bitbucket API rate limit reached, sleeping for 5 sec then retry...");
+            try {
+                Thread.sleep(API_RATE_LIMIT_WAIT_TIME_DEFAULT);
```

Comment:
Just waiting for 5 seconds will cause a lot of polling and spamming of the server. Please shift to some kind of progressive back off where each of retry waits longer (up to some reasonable limit. 

Alternative: see if they the headers of the 429 provide information about when to the limit will reset and sleep for something like that long.